naga/src/front/spv/mod.rs

```diff
@@ -5594,4 +5672,38 @@ mod test {
         ];
         let _ = super::parse_u8_slice(&bin, &Default::default()).unwrap();
     }
+
+    #[test]
+    fn atomic_i_inc() {
```

Comment:
This should become a snapshot test. I think it would be okay to add a `NO_VALIDATION` flag to `Targets` in `naga/tests/snapshots.rs` that directs `check_targets` to return early without trying to validate the module, and then have this test case handled by `convert_spv_all`, passing `Targets::IR | Targets::NO_VALIDATION`. Then you could at least see the IR generated for your SPIR-V.

Reply:
@jimblandy - I'll add this as a snapshot test. Is it ok if I also keep it inline here, as this is an effective "staging ground" for my development workflow. From here I can easily play around with the frontend and the module and I can run `cargo watch -x 'test -p naga -- --nocapture atomic_i_inc'`. Otherwise I'll just keep it in a separate branch.

Reply:
Oh, also - it appears there is no `Targets::NO_VALIDATION` variant. I'm not sure how to keep the snapshot test from failing.

Reply:
Ah, wait - I actually think that after adding this as a snapshot test it's correctly catching the case you mentioned above where the literal used as an increment amount is the wrong type. 👍